src/commands/registration/pasv.js

```diff
@@ -11,12 +11,17 @@ module.exports = {
     this.connector = new PassiveConnector(this);
     return this.connector.setupServer()
     .then((server) => {
-      let address = this.server.options.pasv_url;
-      // Allow connecting from local
-      if (isLocalIP(this.ip)) {
-        address = this.ip;
-      }
       const {port} = server.address();
+      let pasvAddress = this.server.options.pasv_url;
+      if (typeof pasvAddress === "function") {
+        return Promise.try(() => pasvAddress(this.ip))
+          .then((address) => ({address, port}));
+      }
+      // Allow connecting from local
+      if (isLocalIP(this.ip)) pasvAddress = this.ip;
```

Comment:
Let us clarify the intention here - do we want the local IP to be used even if the function is provided? Or would we be assuming that the function would always take that into account? 

Because right now, the return on L17 would skip this logic completely if a function is provided.

Reply:
yes, I just changed the order in the commit, as I assumed that if the user wants to take the control about what IP to return to the FTP client, this should include if remote IP is local or not. If you don't agree I'll revert changes.

Reply:
I think that is a reasonable assumption.